packages/ai/src/data-stream/process-data-stream.ts

```diff
@@ -28,6 +29,7 @@ export async function processDataStream({
   onStartStepPart,
 }: {
   stream: ReadableStream<Uint8Array>;
+  onStreamStart?: () => Promise<void> | void;
```

Comment:
why is this needed? (just curious)

Reply:
for updating status! before we were setting status to streaming every time we processed a chunk